### ISSUE TITLE
AssertionError: Test case is leaking resources

### ISSUE THREAD

**Issue by nadeemramsing** [eveningkid/denodb#164]
Apparently occurs when a resource has not been consumed or closed (see denoland/deno#4735).
_await conn.close()_ did not work.

**Code:**

```
Deno.test("Connection to unavailable db", async () => {
  const conn = new Database("mongo", {
    uri: "mongodb://localhost:27017",
    database: "unavailable-db",
  });

  const ping = await conn.ping();

  await conn.close();

  assertEquals(
    ping,
    false,
    "Database unavailable-db exists or connection to an unavailable db successful",
  );
});
```

**Error:**

AssertionError: Test case is leaking resources.
Before: {
  "0": "stdin",
  "1": "stdout",
  "2": "stderr"
}
After: {
  "0": "stdin",
  "1": "stdout",
  "2": "stderr",
  "3": "tcpStream"
}

Make sure to close all open resource handles returned from Deno APIs before
finishing test case.
    at assert (deno:runtime/rt/06_util.js:33:13)
    at Object.resourceSanitizer [as fn] (deno:runtime/rt/40_testing.js:81:7)
    at async TestRunner.[Symbol.asyncIterator] (deno:runtime/rt/40_testing.js:245:13)
    at async Object.runTests (deno:runtime/rt/40_testing.js:322:22)
    at async file:///mnt/e/Personal%20Projects/deno/denodb/denodb-playground/$deno$test.ts:3:1

**deno --version**
deno 1.6.1 (release, x86_64-unknown-linux-gnu)
v8 8.8.278.2
typescript 4.1.2

**Comment by zepfgeorg** [comment]
I would suggest using [deno_mongo](https://deno.land/x/mongo@v0.20.1) because there the `.close()` function seems to work fine. I had the same problem today and I'm going to switch to it since denodb is using it as its driver anyway so it doesn't make much of a difference.

**Comment by zepfgeorg** [comment]
> I would suggest using [deno_mongo](https://deno.land/x/mongo) because there the `.close()` function seems to work fine. I had the same problem today and I'm going to switch to it since denodb is using it as its driver anyway so it doesn't make much of a difference.

Coming back to this because at least in my configuration, deno_mongo is creating more problems than it solves. Turns out that `MongoClient.close()` was never actually called.

**Comment by eveningkid** [comment]
Hi,

Thanks again Georg for helping out.

#170 should have fixed it.
It's been released under `1.0.19` or `master` depending on how you import denodb.

Feel free to reach out if you get another issue (let's hope not!)
